### PR TITLE
Make control_statement rule substitution correctable

### PR DESCRIPTION
**Pull request by MaxHaertwig**

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Style/ControlStatementRule.swift [hunk 1]
```diff
@@ -80,15 +111,25 @@ public struct ControlStatementRule: ConfigurationProviderRule, AutomaticTestable
...
+        if violationString.contains("(") && violationString.contains(")") {
+            if let openingIndex = violationString.firstIndex(of: "(") {
+                let replacement = violationString[violationString.index(before: openingIndex)] == " " ? "" : " "
+                violationString.replaceSubrange(openingIndex...openingIndex, with: replacement)
+            }
+            if let closingIndex = violationString.lastIndex(of: ")" as Character) {
+                let replacement = violationString[violationString.index(after: closingIndex)] == " " ? "" : " "
+                violationString.replaceSubrange(closingIndex...closingIndex, with: replacement)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (MaxHaertwig) [reply Source/SwiftLintFramework/Rules/Style/ControlStatementRule.swift]: I tried that before, but the closingIndex might change when the first subrange is replaced.

[the remark this answers is not in the document]
Comment (idrougge) [reply Source/SwiftLintFramework/Rules/Style/ControlStatementRule.swift]: How unattentive of me. Please disregard.

File: CHANGELOG.md [hunk 2]
```diff
@@ -71,6 +71,8 @@
 
 * Add GitHub Actions Logging reporter (`github-actions-logging`).  
   [Norio Nomura](https://github.com/norio-nomura)
+* Make `control_statement` rule substitution correctable.
+  [MaxHaertwig](https://github.com/maxhaertwig)
```
Comment (PaulTaykalo) [inline CHANGELOG.md]:
```
* Make `control_statement` rule substitution correctable.  
  [MaxHaertwig](https://github.com/maxhaertwig)
```
It seems that you missed two spaces after the first line.

Comment (MaxHaertwig) [reply CHANGELOG.md]: Thanks for noticing

**Comment by PaulTaykalo** [comment]
Thanks 🎉 . nice one!

**Comment by PaulTaykalo** [comment]
@jpsim I merged it if you don't mind

**Pull request merged**